### ISSUE TITLE
usemin not replacing paths within <!--bower:js -->  blocks

### ISSUE THREAD

**Issue by BirdInTheCity** [yeoman/grunt-usemin#331]
I've used Yeoman to generate my Angular project and have been doing component installation via bower.  

When running `grunt build`, paths within the `bower:js` block are not being replaced with the generated `vendor.js` path link.  If the bower.js block is removed, the substitution happens as expected.  Any thoughts on how I could remedy this?  I've included the relevant grunt config at the bottom.  I've got usemin@2.1.0 installed.

```
index.html
    <!-- build:js scripts/vendor.js -->
    <!-- bower:js -->
    <script src="bower_components/jquery/dist/jquery.js"></script>
    <script src="bower_components/angular/angular.js"></script>
    <script src="bower_components/bootstrap-sass-official/vendor/assets/javascripts/bootstrap.js"></script>
    <script src="bower_components/angular-resource/angular-resource.js"></script>
    <script src="bower_components/angular-cookies/angular-cookies.js"></script>
    <script src="bower_components/angular-sanitize/angular-sanitize.js"></script>
    <script src="bower_components/angular-route/angular-route.js"></script>
    <script src="bower_components/firebase/firebase.js"></script>
    <script src="bower_components/angular-mocks/angular-mocks.js"></script>
    <script src="bower_components/firebase-simple-login/firebase-simple-login.js"></script>
    <script src="bower_components/observe-js/src/observe.js"></script>
    <script src="bower_components/angularfire/angularfire.js"></script>
    <!-- endbower -->
    <!-- endbuild -->

        <!-- build:js({.tmp,app}) scripts/scripts.js -->
        <script src="scripts/app.js"></script>
        <script src="scripts/controllers/main.js"></script>
    <script src="scripts/angularfire/config.js"></script>
    <script src="scripts/services/firebase.js"></script>
    <script src="scripts/services/login.js"></script>
    <script src="scripts/angularfire/simpleLoginTools.js"></script>
    <script src="scripts/angularfire/routesecurity.js"></script>
    <script src="scripts/controllers/login.js"></script>
        <!-- endbuild -->
```

```
Usemin log:

Processing as HTML - dist/index.html
Update the HTML to reference our concat/min/revved script files
<script src="scripts/scripts.js" changed to <script src="scripts/ffc5b2c8.script
s.js"
Update the HTML with the new css filenames
<link rel="stylesheet" href="styles/main.css" changed to <link rel="stylesheet"
href="styles/819d8b60.main.css"
...
```

```
Gruntfile.js
  // Reads HTML for usemin blocks to enable smart builds that automatically
    // concat, minify and revision files. Creates configurations in memory so
    // additional tasks can operate on them
    useminPrepare: {
      html: '<%= yeoman.app %>/index.html',
      options: {
        dest: '<%= yeoman.dist %>',
        flow: {
          html: {
            steps: {
              js: ['concat', 'uglifyjs'],
              css: ['cssmin']
            },
            post: {}
          }
        }
      }
    },

    // Performs rewrites based on rev and the useminPrepare configuration
    usemin: {
      html: ['<%= yeoman.dist %>/{,*/}*.html'],
      css: ['<%= yeoman.dist %>/styles/{,*/}*.css'],
      options: {
        assetsDirs: ['<%= yeoman.dist %>']
      }
    },
```

**Comment by datton** [comment]
Slight thread resurrection -- I had a custom replacement block type from processhtml. It read a list of files from a directory and created script tags before being executed by usemin. The block replacement type had mismatched line endings which causes this similar problem (ie: LF and CRLF mixed). Maybe that'll help someone.

tl;dr: Check line endings
